src/scripts/main.js

```diff
@@ -1 +1,37 @@
-'use strict';
+const track = document.querySelector('.carousel__track');
+const slides = Array.from(track.children);
+const nextButton = document.querySelector('.carousel__button-right');
+const prevButton = document.querySelector('.carousel__button-left');
+
+const slideWidth = slides[0].getBoundingClientRect().width;
+
+const setSlidePosition = (slide, index) => {
+  slide.style.left = slideWidth * index + 'px';
+};
+
+slides.forEach(setSlidePosition);
+
+window.addEventListener('resize', (e) => {
+  slides.forEach(setSlidePosition);
+});
+
+// eslint-disable-next-line no-shadow
+const moveToSlide = (track, currentSlide, targetSlide) => {
+  track.style.transform = 'translateX(-' + targetSlide.style.left + ')';
+  currentSlide.classList.remove('current-slide');
+  targetSlide.classList.add('current-slide');
+};
+
+prevButton.addEventListener('click', (e) => {
+  const currentSlide = track.querySelector('.current-slide');
+  const prevSlide = currentSlide.previousElementSibling;
+
```

Comment:
This gives a console error whenever you scroll to the end of your slides and try to continue after it.
You could add a test here to, in case the slide you're trying to move to does not exist, move it to the last one (like an infinite slider"